### ISSUE TITLE
[1][gibbsfromncis] Account: Contacts Page bugs and improvements

### ISSUE THREAD

**Issue by gibbsfromncis** [bitshares/bitshares-ui#1130]
1. Hide Hidden **Accounts**
- Why `Accounts` if user manage Contacts?  
- Why are them `Hidden` if actually them are Removed?
2. User unable to `undo` his action. User should be able to add contact back by click on the same icon-button (behavior should be like on https://wallet.bitshares.org/#/explorer/accounts) when user can add/delete user by click on the same button
3. The strange `half-dark row` between hidden/visible contacts and on `Accounts` page too. 
4. Hide Hidden Accounts button visible on `Accounts` tab but it's not needed there

![gif-contacts-wrong-behavior](https://user-images.githubusercontent.com/35899973/35932284-98e52778-0c37-11e8-87a6-4c472b345abd.gif)

The improvements/fixes list: (Feel free to update/extend/reduce this list)

- [ ] Rename `Hide/Show hidden Accounts` to `Hide/Show removed Contacts` (On Contacts Tab)
- [x] Add an ability to add contact back by click on `remove` icon. 
- [x] Make `remove` icon as `add` icon when contact is removed (like on [Explorer Accounts](https://wallet.bitshares.org/#/explorer/accounts))
- [x] Remove strange floating row

**Comment by startailcoon** [comment]
When did we change removal of contacts to hide contacts? 

I don't think it makes sense to hide contacts, as you should remove them and then add them in the `Explorer -> Accounts` view if you wish them on your contacts list again.

**Comment by wmbutler** [comment]
I'm not sure what you are trying to accomplish. I agree with your add/remove contact toggle so the row doesn't just disappear, but we need to keep the Settings -> Accounts show / hide pattern in place. This is considered a more permanent way to hide accounts from the front end.

**Comment by gibbsfromncis** [comment]
@wmbutler okay, got it. I didn't know about the "Hide Account" feature on User Settings. Thanks for you feedback.

I've updated the list of changes.

**Comment by startailcoon** [comment]
Please be aware that the menu structure of `Accounts/Contacts/Recent Activity` changes in PR #1143

**Comment by gibbsfromncis** [comment]
@startailcoon cool. Thanks man. Already working on QR scanner for you. Going to prepare it in few hours if everything will be OK.

**Comment by gibbsfromncis** [comment]
@wmbutler 

Unfortunately for now Contacts is using the same API as "Hide/Show" own Accounts. I've created new task to work on it: #1173. It's not necessary but would be a good for feature use of Contacts like on Transaction section. 

Changes I made:
1) Fix divider between Hidden/Shown accounts
2) Do not display divider if hidden accounts doesn't exist
3) Do not display Show/Hide button if hidden accounts doesn't exist
4) Add an ability to add contact back and fix icon
![gif-contacts-fixed](https://user-images.githubusercontent.com/35899973/36237248-fcc28240-11fa-11e8-9b83-d6df2a1a3ee9.gif)

PR: #1174
